dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -245,4 +245,31 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             LOG_IF_FAILED(::CoRevokeClassObject(static_cast<DWORD>(token.Cookie())));
         }
     }
+
+    bool PushNotificationManager::IsBackgroundTaskBuilderAvailable()
+    {
+        auto builder{ BackgroundTaskBuilder() };
+        auto builder5{ builder.try_as<winrt::IBackgroundTaskBuilder5>() };
+        return builder5 != nullptr;
+    }
+
+    bool PushNotificationManager::IsActivatorSupported(PushNotificationRegistrationOptions const& options)
+    {
+        THROW_HR_IF(E_INVALIDARG, options == PushNotificationRegistrationOptions::Undefined); // WI_IsFlagSet does not allow 0 as flag.
+
+        bool isBackgroundTaskFlagSet{ WI_IsAnyFlagSet(options, PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator) };
+        bool isProtocolActivatorSet{ WI_IsFlagSet(options, PushNotificationRegistrationOptions::ProtocolActivator) };
+
+        THROW_HR_IF(E_INVALIDARG, IsBackgroundTaskBuilderAvailable && isProtocolActivatorSet); // Invalid flag combination
+
+        if (AppModel::Identity::IsPackagedProcess() && IsBackgroundTaskBuilderAvailable())
+        {
+            THROW_HR_IF(E_INVALIDARG, isProtocolActivatorSet); // Don't allow packaged apps to call with protocol if background task available
```

Comment:
Why throw? Instead simply return false because protocol activation is not supported if Background task is available.